client/state/comments/actions.js

```diff
@@ -0,0 +1,381 @@
+		} )
+		.catch( ( error ) => {
+			dispatch( {
+				type: COMMENTS_ERROR_COMMENT,
+				siteId,
+				postId,
+				commentId: placeholderComment.ID,
+				error
+			} );
+
+			throw error;
+		} );
+	};
+}
+
+/***
+ * Fetches all comments with given fields, default only ID
+ * @param {Number} siteId site identifier
+ * @param {Number} postId post identifier
+ * @param {String[]} additionalFields field names array
+ * @returns {Promise<Object[]>} promise of array of comments with specified fields
+ */
+export function fetchAllCommentFields( siteId, postId, additionalFields = [], numberOfItemsPerBatch = MAX_NUMBER_OF_COMMENTS_IN_FETCH_RESULT ) {
+	const fields = [ 'ID' ].concat( additionalFields );
+	const localFieldsMapper = fieldsMapper.bind( fieldsMapper, fields );
+
+	// Default order is DESC, so we're omitting it
+	// https://developer.wordpress.com/docs/api/1.1/get/sites/%24site/posts/%24post_ID/replies/
+	const query = {
+		fields: fields.join( ',' ),
+		number: numberOfItemsPerBatch
+	};
+
+	// we're doing one non-batched request to get the comments count,
+	// so we'll know how many batched requests to perform
+	return wpcom.site( siteId )
+		.post( postId )
+		.comment()
+		.replies( query )
+		.then( ( { found, comments } ) => {
+			const commentIds = comments.map( localFieldsMapper );
+
+			if ( found > numberOfItemsPerBatch ) {
+				const batch = wpcom.batch();
+				const batchUrls = [];
+
+				for ( let i = numberOfItemsPerBatch; i < found; i += numberOfItemsPerBatch ) {
+					batchUrls.push( createUrlFromTemplate( '/sites/$site/posts/$post_ID/replies/', { site: siteId, post_ID: postId }, Object.assign( {}, query, { offset: i } ) ) );
+				}
+
+				batchUrls.forEach( batch.add.bind( batch ) );
+
+				return batch.run().then( ( batchResultForUrls ) => {
+					let arrayOfCommentIdsArrays = batchUrls.map( ( batchUrl ) => batchResultForUrls[ batchUrl ].comments )
+															.map( ( arrayOfComments ) => arrayOfComments.map( localFieldsMapper ) );
+
+					return Array.prototype.concat.apply( commentIds, arrayOfCommentIdsArrays );
+				} );
+			}
+
+			return commentIds;
+		} );
+}
+
+/***
+ * Fetches comments by ids
+ * @param {Number} siteId site identifier
+ * @param {Number[]} commentIds array of comment identifiers
+ * @returns {Promise<Object[]>} promise of array of comments
+ */
+export function fetchCommentsByIds( siteId, commentIds ) {
+	if ( ! siteId ) {
+		throw new Error( 'No siteId supplied' );
+	}
+
+	if ( ! Array.isArray( commentIds ) || commentIds.length < 1 ) {
+		throw new Error( 'At least a single commentId should be supplied' );
+	}
+
+	const batch = wpcom.batch();
+	const batchUrls = commentIds.map( ( commentId ) => createUrlFromTemplate( '/sites/$site/comments/$comment_ID', { site: siteId, comment_ID: commentId } ) );
+
+	batchUrls.forEach( batch.add.bind( batch ) );
+
+	return batch.run().then( ( batchResultForUrls ) => {
+		let arrayOfcomments = batchUrls.map( ( batchUrl ) => batchResultForUrls[ batchUrl ] );
```

Comment:
@blowery I've addressed it, thanks for pointing that out.

[Here](https://github.com/Automattic/wp-calypso/pull/3143/files#diff-a1e5174c2173e0927694ec3bd26d79d4R243) and [here](https://github.com/Automattic/wp-calypso/pull/3143/files#diff-a1e5174c2173e0927694ec3bd26d79d4R282).